Improve rollback cleanup

pirat89 opened the pull request:
- removes files of target kernel when --clean option is used after the rollback
- remove /boot/rollback and /boot/manualcleanup dirs when --clean is used
- add /boot/rollback/do_rollback script that will do rollback from RHEL 7

pirat89 committed:
Remove files of the target kernel under /boot

After the rollback, kernel / bootloader files of target kernel have
been kept in the /boot parition. As easier solution of this,
I put cleaning of this file automatically when the --clean option is
used. As we require run of r-u-t with the --clean option after the
rollback, it is for now safe solution.

To ensure that we remove only files that are created during the
upgrade and will not remove kernel files that it is available
under dual boot (unexpected, but just to be sure), it is checked
version of target kernel and in case `kernelpath` of this version
doesn't exist already, information is stored under `rollback_dir`
in the `.target-kernel` file. Only in case the data is stored in the
file, kernel files of target kernel will be cleaned.

pirat89 committed:
Generate script file that will do rollback from upgraded OS

Previous user experience was not so good as user has to run:
  # cd /boot && python -m rollback.system_restore

Now just needs to run the command below to do a rollback:
  # /boot/rollback/do_rollback

Fixes: #43

pirat89 committed:
Cleanup of rollback dirs in /boot

- clean rut dirs on boot partition by the cleanup script
- clean rut dirs on boot when the --clean option is used
- cleanup script: do not process anything on import
  In case of import of the module automatically cleanup is processed
  but it should be processed only on call.

pirat89 commented:
~~I apologize that I havn't time to test it. So in case you find an issue, feel free to fix it or wait to my pre-testing. But I will be really glad for help here as I have rush days and will be able to test it properly tomorrow. Thanks~~

pirat89 committed:
Add executor for manual cleanup: /boot/rollback/do_cleanup

pirat89 committed:
Fix chmod and rollback import

When os.chmod is used, use of 0xxx mode is incorrect. Either xxx
is wrong as mode is not set as expected. Only correct way here is
use of 0oxxx format. E.g.:
    0o775

Additionaly, current solution around rollbacks directly from RHEL 7
system really needs special run and it requires different import
when it is used on RHEL 6 and when it is used on RHEL 7 to provide
a rollback or cleanup.

pirat89 committed:
Define important paths just once (for rollback)

Previously, important paths has been defined on several places.
Not it should be defined only once. The `/boot/manualcleanup`
directory has been omitted and all content is stored uder the
`rollback_dir`.

mfranczy commented on redhat-upgrade-tool.py:
```diff
@@ -316,6 +319,21 @@ def main(args):
+            kv = kernelver(kernel)
+            if not kv:
+                print _("Warning: cannot determine version of target kernel."
+                        " In case of rollback, you will need to removed kernel"
```
typo, fix to ```you will need to remove kernel``` or ```you will have to```

mfranczy commented on redhat_upgrade_tool/rollback/__init__.py:
```diff
@@ -1,2 +1,10 @@
+active_kernel_file = os.path.join(rollback_dir, '.active-kernel')
+all_kernels_file = os.path.join(rollback_dir, '.all-kernels')
+target_kernel_file = os.path.join(rollback_dir, '.target-kernel')
+snap_boot_files_file = os.path.join(rollback_dir, '.snap_boot_files')
```
I would change the name of this variable, ```snap_boot_files_file```, I see why it's like this, although if someone is looking after a long time into that code, they might be confused.

But it is only a suggestion, up to you..

pirat89 replied:
I agree and I spent a time above this name already. But I still can't come up with any good name at all, soI will keep this for now.

mfranczy commented:
I added two minor things, it should work (didn't test it yet).

pirat89 commented:
Related #45

pirat89 committed:
Clean grub2 & kdump of target kernel

In case /boot/grub2 doesn't exist before the upgrade (expected),
in case of rollback the directory will be removed. Additionally
remove the kdump of target kernel to save space on boot partition.
